pkg/agent/plugin/nodeattestor/v0.go

```diff
@@ -0,0 +1,88 @@
+package nodeattestor
+
+import (
+	"context"
+	"io"
+
+	"github.com/spiffe/spire/pkg/common/catalog"
+	nodeattestorv0 "github.com/spiffe/spire/proto/spire/agent/nodeattestor/v0"
+)
+
+// V0 instantiates a NodeAttestor for a v0 versioned plugin
+func V0(pluginInfo catalog.PluginInfo, client nodeattestorv0.NodeAttestor) NodeAttestor {
+	return nodeAttestorV0{
+		// This is a little gross for now. This will be cleaner when the new
+		// catalog is introduced.
+		PluginFacade: catalog.PluginFacade{PluginInfo: pluginInfo},
+		client:       client,
+	}
+}
+
+type nodeAttestorV0 struct {
+	catalog.PluginFacade
+	client nodeattestorv0.NodeAttestor
+}
+
+func (v0 nodeAttestorV0) Attest(ctx context.Context, serverStream ServerStream) error {
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+
+	pluginStream, err := v0.client.FetchAttestationData(ctx)
+	if err != nil {
+		return v0.WrapErr(err)
```

Comment:
The idea here is that all errors flowing out of the plugin go through the `catalog.PluginFacade` methods to ensure that they are properly prefixed.

catalog.PluginFacade only applies the prefix if not already present. This is temporary and in the future it will just apply the prefix, but we need to maintain backwards compatibility with plugin implementations that are applying the prefix themselves.

WrapErr is used to wrap generic errors that the shim doesn't care to interpret. InternalError(f) is used to produce an error when the shim detects bad plugin behavior (i.e. the plugin isn't conforming to the contract).